llama_index/llms/huggingface.py

```diff
@@ -294,3 +306,181 @@ def stream_chat(
+            " Defaults to None, meaning it will loop until the server is available."
+        ),
+    )
+    headers: Dict[str, str] = Field(
+        default=None,
+        description=(
+            "Additional headers to send to the server. By default only the"
+            " authorization and user-agent headers are sent. Values in this dictionary"
+            " will override the default values."
+        ),
+    )
+    cookies: Dict[str, str] = Field(
+        default=None, description="Additional cookies to send to the server."
+    )
+    _sync_client: "InferenceClient" = PrivateAttr()
+    _async_client: "AsyncInferenceClient" = PrivateAttr()
+
+    def _get_inference_client_kwargs(self) -> Dict[str, Any]:
+        """Extract the Hugging Face InferenceClient construction parameters."""
+        return {
+            "model": self.model_name,
+            "token": self.token,
+            "timeout": self.timeout,
+            "headers": self.headers,
+            "cookies": self.cookies,
+        }
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize.
+
+        Args:
+            kwargs: See the class-level Fields.
+        """
+        super().__init__(**kwargs)  # Populate pydantic Fields
+        try:
+            from huggingface_hub import AsyncInferenceClient, InferenceClient
+        except ModuleNotFoundError as exc:
+            raise ImportError(
+                f"{type(self).__name__} requires huggingface_hub with its inference"
+                " extras, please run `pip install huggingface_hub[inference]`."
+            ) from exc
+        self._sync_client = InferenceClient(**self._get_inference_client_kwargs())
+        self._async_client = AsyncInferenceClient(**self._get_inference_client_kwargs())
+
+    def validate_supported(self, task: str) -> None:
+        """Confirm the contained model_name is deployed on the Inference API service."""
+        all_models = self._sync_client.list_deployed_models(frameworks="all")
+        try:
+            if self.model_name not in all_models[task]:
+                raise ValueError(
+                    f"The Inference API service doesn't have the model"
+                    f" {self.model_name!r} deployed."
+                )
+        except KeyError as exc:
+            raise KeyError(
+                f"Input task {task!r} not in possible tasks {list(all_models.keys())}."
+            ) from exc
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        raise NotImplementedError
+
+    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
+        return conversational_output_to_chat_response(
+            output=self._sync_client.conversational(
+                **{**chat_messages_to_conversational_kwargs(messages), **kwargs}
+            )
+        )
+
+    def complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
+        return CompletionResponse(
```

Comment:
What is `raw` and `delta` used for in `CompletionResponse`?

Should I be using them here?

Reply:
`raw` is the raw json response from the API (if any)

`delta` is used in streaming, it contains whatever new text was just streamed. Check out this for an example
https://github.com/run-llama/llama_index/blob/979b4332854f608b7f1a22740e1f84e55c1d3077/llama_index/llms/huggingface.py#L280

Reply:
Thank you!